src/item.rs

```diff
@@ -723,3 +738,165 @@ impl<F> Invoke<P<ast::Ty>> for ItemTyBuilder<F>
         self.build_ty(ty)
     }
 }
+
+//////////////////////////////////////////////////////////////////////////////
+
+pub struct ItemImplBuilder<F> {
+    builder: ItemBuilder<F>,
+    unsafety: ast::Unsafety,
+    polarity: ast::ImplPolarity,
+    generics: ast::Generics,
+    trait_ref: Option<ast::TraitRef>,
+    items: Vec<P<ast::ImplItem>>,
+}
+
+impl<F> ItemImplBuilder<F>
+    where F: Invoke<P<ast::Item>>,
+{
+    pub fn unsafe_(mut self) -> Self {
+        self.unsafety = ast::Unsafety::Unsafe;
+        self
+    }
+
+    pub fn negative(mut self) -> Self {
+        self.polarity = ast::ImplPolarity::Negative;
+        self
+    }
+
+    pub fn with_generics(mut self, generics: ast::Generics) -> Self {
+        self.generics = generics;
+        self
+    }
+
+    pub fn generics(self) -> GenericsBuilder<Self> {
+        GenericsBuilder::new_with_callback(self)
+    }
+
+    pub fn with_trait(mut self, trait_ref: ast::TraitRef) -> Self {
+        self.trait_ref = Some(trait_ref);
+        self
+    }
+
+    pub fn trait_(self) -> ImplTraitBuilder<Self> {
```

Comment:
I thought about that.  Few things:

1. Was thinking ahead that if there's another Path-able component to an Impl, then there would be a conflict of callbacks.
2. The main reason is that it would require an extra `.build()` call to finalize the Path and callback to the ItemImplBuilder.  Most of the other builders build implicitly on calls to `id()` so this is consistent with that.